docs/external.rst

```diff
@@ -1,68 +1,140 @@
 .. _external:
 
-Interacting with external arrays
-================================
+Exporting and importing data
+============================
 
-**External arrays** refer to ``numpy.ndarray`` or ``torch.Tensor``.
+Introduction
+------------
 
-Conversion between Taichi fields and external arrays
-----------------------------------------------------
+Taichi fields are aimed to be used in Taichi-scope.
+However, sometimes we may need to access Taichi field elements in Python-scope,
+which is not very efficient when a lot of elements within the field are required.
 
-Use ``to_numpy``/``from_numpy``/``to_torch``/``from_torch``:
+So we provide a way to export and import data in Taichi fields via NumPy arrays.
+This allows you to interact with the data in Python-scope with other packages.
```

Comment:
```suggestion
Therefore, we provide handy functions for flexible conversion between Taichi fields, NumPy arrays, and PyTorch tensors.
```

Reply:
`between Taichi fields, NumPy arrays, and PyTorch tensors` sounds confusing to me.
It can increase learning barrier, beginners may ask: What is NumPy and PyTorch? Which is the best way for me?

If we say `export and import data in Taichi fields via NumPy arrays.`.
Beginners will say: Oh I see, NumPy seems to be a hard dep of Taichi, let me learn NumPy first!

---

Note that PyTorch is an optional dependency that is not used by most of end-users.
We don't provide `to_blender`  despite `taichi` could be useful in Blender.
NumPy will stay "superior" since I believe PyTorch and Blender do provide NumPy interface too.

My idea is: **As lone as a package provides NumPy interface**, we don't need to customize a `to_xxx` for it.
`to_torch` is just for decreasing memory overhead IMO.

Reply:
OK, `to_torch` is here to stay, but given that it's aimed to reduce memory overhead, let's do not emphasis it too much in a section faced to beginners.